pkg/kds/global/components.go

```diff
@@ -123,19 +103,12 @@ func Setup(rt runtime.Runtime) error {
 			kds_sync_store_v2.GlobalSyncCallback(stream.Context(), resourceSyncerV2, rt.Config().Store.Type == store_config.KubernetesStore, kubeFactory, rt.Config().Store.Kubernetes.SystemNamespace),
 			rt.Config().Multizone.Global.KDS.ResponseBackoff.Duration,
 		)
-		go func() {
-			err := sink.Receive()
-			if err != nil && (status.Code(err) != codes.Canceled && !errors.Is(err, context.Canceled)) {
-				err = errors.Wrap(err, "KDSSyncClient finished with an error")
-				select {
-				case errChan <- err:
-				default:
-					log.Error(err, "failed to write error to closed channel")
-				}
-			} else {
-				log.V(1).Info("KDSSyncClient finished gracefully")
-			}
-		}()
+		if err := sink.Receive(); err != nil && (status.Code(err) != codes.Canceled && !errors.Is(err, context.Canceled)) {
```

Comment:
Hmm why is this necessarily OK to run synchronously?

Reply:
This callback already runs asynchronously like:
```go
processingErrorsCh := make(chan error, 1)
go func() {
	if err := g.zoneToGlobalCb.OnZoneToGlobalSyncConnect(stream); err != nil {
		processingErrorsCh <- err
	}
}()
```
so I don't really understand why `sink.Receive` had to run asynchronously as well

Reply:
OK so it's fine, since we don't wait for it to return anyway